erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -118,6 +121,11 @@ def set_bank_account_data(self):
 			if not self.get(field):
 				self.set(field, bank_data.account)
 
+	def validate_payment_type_with_outstanding(self):
+		total_outstanding = sum(d.allocated_amount for d in self.get('references'))
+		if total_outstanding < 0 and self.party_type == 'Customer' and self.payment_type == 'Receive':
+			frappe.throw(_("Cannot receive from customer against negative outstanding"), title=_("Incorrect Payment Type"))
```

Comment:
We should allow receiving payments, but in that payment entry earlier credit notes should not be adjusted

Reply:
I have not blocked receiving payments, I didn't completely understand this. Can you please elaborate